src/sensordatainterface/views/edit_views.py

```diff
@@ -1220,7 +1221,7 @@ def edit_retrieval(request, deployment_id=None, retrieval_id=None):
 
     if request.method == 'POST':
         updating = request.POST['action'] == 'update'
-        deployment_action = Action.objects.get(pk=request.POST['deploymentaction'])
+        deployment_action = Action.objects.get(pk=request.POST.get('deployment_id'))

```

Comment:
careful with this. make sure it's the right parameter for all cases if there are more cases in which this view is used.